### PR TITLE
Transform fetch errors to asserts.

### PR DESCRIPTION
**Pull request by cramforce**
So, they go into our user error bucket.

### CODE REVIEW

**Comment by cramforce** [comment]
Please check the boolean logic!

**Comment by dvoytenko** [comment]
> they go into out user error bucket.

They go "into" or "out of" user error bucket?

File: src/xhr.js [hunk 1]
```diff
@@ -89,26 +89,25 @@ class Xhr {
...
-      throw new Error(`Source origin is not allowed in ${input}`);
-    } else {
-      input = addParamToUrl(input, SOURCE_ORIGIN_PARAM, sourceOrigin);
-    }
-    return this.fetch_(input, opt_init).then(response => {
+    assert(!(SOURCE_ORIGIN_PARAM in query), 'Source origin is not allowed in %s',
+        input);
+    input = addParamToUrl(input, SOURCE_ORIGIN_PARAM, sourceOrigin);
+    return this.fetch_(input, opt_init).catch(reason => {
+      assert(false, 'Fetch failed %s: %s', input, reason.message);
```
Comment (dvoytenko) [inline src/xhr.js]: `reason.message` is not guaranteed. Are you ok with this here? Also, we loose the stack in this case, don't we?

Comment (cramforce) [inline src/xhr.js]: I think loosing the stack is fine. This is the inner most promise.

**Comment by dvoytenko** [comment]
LGTM

**Pull request merged**